### DISCUSSION TITLE
State not updating inside table instance

### DISCUSSION THREAD

**Discussion by MarleyDabb** [TanStack/table#4049] [Q&A]
Hi, I'm using the new react table and I've noticed my state variables aren't updating inside the table.

```
 const columns = useMemo(() => [
    table.createGroup({
      header: `Cashflow Table - ${selectedScenario.name}`,
      columns: [
        .....
```

The selectScenario value is set and changed using react state, however when I change the value it doesn't update inside the table. I tried placing the variable above the table and it updates fine, though in the table it stays the same and doesnt change. Any idea why?

**Comment by tannerlinsley** [comment]
Got a code sandbox? That would help debug
